File: typescript/examples/send-many.mjs
// Backfill history with silent=true (stored, nobody notified), 4 requests in flight.
import { SignalClient } from "@tokenearly/signal";

const token = process.env.TOKENEARLY_SIGNAL_TOKEN;
if (!token) {
  console.error("set TOKENEARLY_SIGNAL_TOKEN first");
  process.exit(1);
}

const history = [
  { id: "2026-09-01-btc", en: "BTC closed above 65000", zh: "BTC 收于 65000 上方", ts: 1756742400 },
  { id: "2026-09-02-eth", en: "ETH closed above 3200", zh: "ETH 收于 3200 上方", ts: 1756828800 },
];

const client = new SignalClient(token);
const batch = await client.sendMany(
  history.map((item) => ({
    source_id: "example_signal",
    event_id: `backfill-${item.id}`,
    titles: { zh: item.zh, en: item.en },
    published_at: item.ts,
    silent: true,
  })),
  { concurrency: 4 },
);

console.log(`sent=${batch.sent} failed=${batch.failed}`);
for (const [index, error] of batch.errors) console.log(`  #${index}: ${error.message}`);

File: python/examples/send_many.py
"""Backfill historical events with silent=True (stored, nobody notified)."""
import os

from tokenearly_signal import Event, SignalClient

token = os.environ.get("TOKENEARLY_SIGNAL_TOKEN") or exit("set TOKENEARLY_SIGNAL_TOKEN first")

history = [
    {"id": "2026-09-01-btc", "title_en": "BTC closed above 65000", "title_zh": "BTC 收于 65000 上方", "ts": 1756742400},
    {"id": "2026-09-02-eth", "title_en": "ETH closed above 3200", "title_zh": "ETH 收于 3200 上方", "ts": 1756828800},
]

events = [
    Event(
        source_id="example_signal",
        event_id=f"backfill-{item['id']}",
        titles={"zh": item["title_zh"], "en": item["title_en"]},
        published_at=item["ts"],
        silent=True,
    )
    for item in history
]

with SignalClient(token) as client:
    batch = client.send_many(events)

print(f"sent={batch.sent} failed={batch.failed}")
for index, error in batch.errors.items():
    print(f"  event #{index} ({events[index].event_id}): {error}")
